source/appModules/soffice.py

```diff
@@ -340,3 +340,30 @@ def event_NVDAObject_init(self, obj):
 			# This is a word processor document.
 			obj.description = None
 			obj.treeInterceptorClass = SymphonyDocument
+
+	def searchStatusBar(self, obj, max_depth=5):
+		"""Searches for and returns the status bar object
+		if either the object itself or one of its recursive children
+		(up to the given depth) has the corresponding role."""
+		if obj.role == controlTypes.Role.STATUSBAR:
+			return obj
+		if max_depth < 1 or obj.role not in [controlTypes.Role.ROOTPANE, controlTypes.Role.WINDOW]:

```

Comment:
A set is probably more appropriate here, not that it is a big deal

```suggestion
		if max_depth < 1 or obj.role not in {controlTypes.Role.ROOTPANE, controlTypes.Role.WINDOW}:
```